waves-integration/src/main/scala/com/wavesplatform/dex/grpc/integration/clients/async/WavesBlockchainAsyncClient.scala

```diff
@@ -12,6 +17,26 @@ import monix.reactive.Observable
 object WavesBlockchainAsyncClient {
   type SpendableBalance        = Map[Asset, Long]
   type SpendableBalanceChanges = Map[Address, SpendableBalance]
+
+  private val briefWavesDescription =
+    Option(
+      BriefAssetDescription(
+        name = ByteStr("WAVES".getBytes(StandardCharsets.UTF_8)),
+        decimals = 8,
+        hasScript = false
+      ))
+
+  final implicit class Ops[F[_]: Functor: Applicative](val self: WavesBlockchainAsyncClient[F]) {
+    def assetDescription(asset: Asset): F[Option[BriefAssetDescription]] =
+      asset.fold(Applicative[F].pure(briefWavesDescription))(self.assetDescription)
+
+    def assetDecimals(asset: Asset.IssuedAsset): F[Option[Int]] = self.assetDescription(asset).map { _.map(_.decimals) }
+
+    // TODO
+    def assetDecimals(asset: Asset): F[Option[Int]] = asset.fold { Applicative[F].pure(Option(8)) } { issuedAsset =>
```

Comment:
Use function from line 33?

Reply:
It's used in a case of an issued asset.

Reply:
I mean use frunction from line 33 as argument in the second arguments list in `fold`